packages/nvidia_nat_core/tests/nat/utils/test_atif_converter.py

```
@@ -361,6 +395,157 @@ def test_serialization_roundtrip(
+
+    def test_tool_call_id_and_observation_source_match(
+        self,
+        batch_converter: IntermediateStepToATIFConverter,
+        simple_trajectory: list[IntermediateStep],
+    ):
+        """tool_call_id and observation source_call_id are linked correctly."""
+        result = batch_converter.convert(simple_trajectory)
+        agent_with_tools = result.steps[1]
+        assert agent_with_tools.tool_calls is not None
+        assert agent_with_tools.observation is not None
+        for tc, obs in zip(agent_with_tools.tool_calls, agent_with_tools.observation.results):
```

Ruff (0.15.6)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)